### PR TITLE
fix: Add prefix in userData

### PR DESCRIPTION
**Pull request by bertux**
## Description of the changes
To be able to filter SuperFluid subgraph events containing only this prefix

### CODE REVIEW

**Review by kevindavee — APPROVED** [review]

File: packages/payment-processor/src/payment/erc777-stream.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export async function payErc777StreamRequest(
     flowRate: expectedFlowRate ?? '0',
     receiver: paymentAddress,
     superToken: superToken.address,
-    userData: `0x${paymentReference}`,
+    userData: `0xbeefac${paymentReference}`,
```
Comment (AdamDowson) [inline packages/payment-processor/src/payment/erc777-stream.ts:54]: Can you add a comment or a link to the docs as to why we need this?

**Review by KolevDarko — APPROVED** [review]

File: packages/payment-processor/src/payment/erc777-stream.ts (hunk 1 above)
Comment (bertux) [reply packages/payment-processor/src/payment/erc777-stream.ts:54]: Yes, I will add a comment with the text I've put in the description of the changes in this pull request.

**Comment by bertux** [comment]
I still need to add this prefix in the queries of payment-detection package

File: packages/payment-processor/src/payment/erc777-stream.ts [hunk 2]
```diff
@@ -47,11 +47,12 @@ export async function payErc777StreamRequest(
   // - use expectedStartDate to compute offset between start of invoicing and start of streaming
   // - start fee streaming
   const { paymentReference, paymentAddress, expectedFlowRate } = getRequestPaymentValues(request);
+  // To be able to filter SuperFluid subgraph events containing only this prefix
   const streamPayOp = sf.cfaV1.createFlow({
     flowRate: expectedFlowRate ?? '0',
     receiver: paymentAddress,
     superToken: superToken.address,
-    userData: `0x${paymentReference}`,
+    userData: `0xbeefac${paymentReference}`,
```
Comment (alexandre-abrioux) [inline packages/payment-processor/src/payment/erc777-stream.ts:55]: Can you add a comment to explain how this prefix was generated? Is this just random or something that we normalized?

Comment (bertux) [reply packages/payment-processor/src/payment/erc777-stream.ts:55]: I'm documenting it in the specs branch. For now I've just generated a prefix that's I've not seen yet in other streams events and I've tried to normalize it as some meaningful text using hexadecimal letters.

Comment (alexandre-abrioux) [reply packages/payment-processor/src/payment/erc777-stream.ts:55]: Great! Thanks

**Review by yomarion — APPROVED** [review]

File: packages/payment-processor/src/payment/erc777-stream.ts (hunk 2 above)
Comment (bertux) [reply packages/payment-processor/src/payment/erc777-stream.ts:57]: documented at https://github.com/RequestNetwork/requestNetwork/pull/688/commits/90e13c495929617a0e7091714515b09ae7c3d294

**Pull request merged**